src/compiler/parser.cc

```diff
@@ -937,6 +937,20 @@ Declaration* Parser::parse_declaration(bool is_abstract) {
   }
   auto parameters = return_type_parameters.second;
 
+  // Allow colons to be at the same level as the declaring method.
+  // Something like:
+  //   foo
+  //       param1
+  //       param2
+  //   :
+  //     body
+  if (current_token() == Token::DEDENT) {
+    if (peek_token() == Token::COLON &&
+        indentation_stack_.top_indentation() == current_indentation() &&
+        indentation_stack_.is_outmost(IndentationStack::DECLARATION_SIGNATURE)) {
+      consume();
+    }
+  }
   Sequence* body;
   if (current_token() == Token::COLON) {
```

Comment:
Doesn't this allow too much?

```
foo
    x
::
  return x
```

Reply:
It only consumes the DEDENT token.
Added a comment.
done.

Reply:
Thanks!